src/repository-utils.ts

```diff
@@ -0,0 +1,143 @@
+import execa from 'execa';
+import path from 'path';
+
+import { ONE_HOUR } from './constants';
+import { createModuleLogger, projectLogger } from './logging-utils';
+import { getEntryStats } from './misc-utils';
+
+const log = createModuleLogger(projectLogger, 'existing-repository');
+
+/**
+ * Information about a Git branch.
+ */
+export type BranchInfo = {
+  currentBranchName: string;
+  defaultBranchName: string;
+  lastFetchedDate: Date | null;
+};
+
+/**
+ * Collects the current and default branch name of the given repository as well
+ * as the time when commits were last fetched.
+ *
+ * @param repositoryDirectoryPath - The path to the repository.
+ */
+export async function getBranchInfo(
+  repositoryDirectoryPath: string,
+): Promise<BranchInfo> {
+  const currentBranchName = await getCurrentBranchName(repositoryDirectoryPath);
+  const defaultBranchName = await getDefaultBranchName(repositoryDirectoryPath);
+  const lastFetchedDate = await getLastFetchedDate(repositoryDirectoryPath);
+  return {
+    currentBranchName,
+    defaultBranchName,
+    lastFetchedDate,
+  };
+}
+
+/**
+ * Retrieves the name of the branch that the given repository is currently
+ * pointing to (i.e., HEAD).
+ *
+ * @param repositoryDirectoryPath - The path to the repository.
+ * @returns The name of the current branch.
+ * @throws If HEAD is not pointing to a branch, but rather an arbitrary commit.
+ */
+export async function getCurrentBranchName(
+  repositoryDirectoryPath: string,
+): Promise<string> {
+  log('Running: git symbolic-ref --quiet HEAD');
+  const { stdout } = await execa('git', ['symbolic-ref', '--quiet', 'HEAD'], {
+    cwd: repositoryDirectoryPath,
+  });
+  const match = stdout.match(/^refs\/heads\/(.+)$/u);
+  const currentBranchName = match?.[1] ?? null;
+  if (!currentBranchName) {
+    throw new Error(
+      `The repository '${repositoryDirectoryPath}' does not seem to be on a branch. Perhaps HEAD is detached? Either way, you will need to return this repo to the default branch manually.`,
+    );
+  }
+  return currentBranchName;
+}
+
+/**
+ * Retrieves the default branch of the given repository, that is, the branch
+ * that represents the main line of development. Unfortunately there's no good
```

Comment:
We can get it from `gh repo view --json defaultBranchRef [repo]`

Reply:
Ah! GitHub _would_ know, wouldn't they... This is fantastic.

Reply:
Added in [ffbdc89](https://github.com/MetaMask/module-lint/pull/30/commits/ffbdc89a90591fd94073c7550b5e8e26269df766).